File: fitguard-core/src/models/InjuryLog.js
const mongoose = require('mongoose');

const injuryLogSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  muscleGroup: {
    type: String,
    required: true,
    trim: true,
    lowercase: true
  },
  injuryType: {
    type: String,
    required: true,
    trim: true,
    lowercase: true
  },
  severity: {
    type: String,
    enum: ['mild', 'moderate', 'severe'],
    required: true
  },
  dateOccurred: {
    type: Date,
    required: true
  },
  recoveryStatus: {
    type: String,
    enum: ['active', 'recovered'],
    default: 'active',
    required: true
  },
  notes: {
    type: String,
    default: ''
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('InjuryLog', injuryLogSchema);

File: lib/features/injuries/data/models/injury_log.dart
class InjuryLog {
  final String id;
  final String muscleGroup;
  final String injuryType;
  final String severity;
  final DateTime dateOccurred;
  final String recoveryStatus;

  InjuryLog({
    required this.id,
    required this.muscleGroup,
    required this.injuryType,
    required this.severity,
    required this.dateOccurred,
    required this.recoveryStatus,
  });

  factory InjuryLog.fromJson(Map<String, dynamic> json) {
    return InjuryLog(
      id: json['_id'] ?? '',
      muscleGroup: json['muscleGroup'] ?? '',
      injuryType: json['injuryType'] ?? '',
      severity: json['severity'] ?? '',
      dateOccurred: DateTime.parse(json['dateOccurred']),
      recoveryStatus: json['recoveryStatus'] ?? '',
    );
  }
}
